optuna/samplers/_tpe/parzen_estimator.py

```diff
@@ -145,6 +145,18 @@ def log_pdf(self, samples_dict: Dict[str, np.ndarray]) -> np.ndarray:
         n_samples = next(iter(samples_dict.values())).size
         if n_samples == 0:
             return np.asarray([], dtype=float)
+
+        # When the search space is one CategoricalDistribution, we use the faster processing,
+        # whose computation result is equivalent to the general one.
+        if len(self._search_space.items()) == 1:
+            param_name, dist = list(self._search_space.items())[0]
+            if isinstance(dist, distributions.CategoricalDistribution):
+                samples = samples_dict[param_name]
+                categorical_weights = self._categorical_weights[param_name]
+                assert categorical_weights is not None
+                ret = np.log(np.inner(categorical_weights.T, self._weights))[samples]
+                return ret
```

Comment:
[nits] My idea is adding the following code at line 169. Which one is better? 🤔

```
if len(self._search_space.items()) == 1:
    ret = np.log(np.inner(categorical_weights.T, self._weights))[samples]
    return ret
```

Reply:
Yes, it is equivalent. But I prefer to separate the code for the exceptional cases from the general one in terms of readability.